File: orion-viewer/src/main/java/com/artifex/mupdfdemo/TextWord.java
package com.artifex.mupdfdemo;

import android.graphics.RectF;

public class TextWord extends RectF {
	private StringBuilder w;

	public TextWord() {
		super();
		w = new StringBuilder();
	}

	public void Add(TextChar tc) {
		super.union(tc);
		w = w.append(tc.c);
	}

    public StringBuilder getText() {
        return w;
    }

    public int textLength() {
        return w.length();
    }
}

File: orion-viewer/src/main/java/universe/constellation/orion/viewer/djvu/TextWord.kt
package universe.constellation.orion.viewer.djvu

import android.graphics.Rect

class TextWord {

    private var stringBuilder = StringBuilder()

    val rect = Rect()

    fun add(value: String, rect: Rect) {
        stringBuilder.append(value)
        this.rect.union(rect)
    }

    fun isNotEmpty(): Boolean {
        return stringBuilder.isNotEmpty()
    }

    fun width(): Int {
        return rect.width()
    }

    fun height(): Int {
        return rect.height()
    }

    override fun toString(): String {
        return stringBuilder.toString()
    }
}